src/catsDb.js

```diff
@@ -1,38 +1,72 @@
 
-const cats = [
-    {
-        id: 1,
-        name: "Catty",
-        sex: "F",
-        coat: "medium hair",
-        description: "buff or tan, skinny, talkative, often found in tall grasses or deep in the bush",
-        breedId: 1
-    },
-    {
-        id: 2,
-        name: "Frank",
-        sex: "M",
-        coat: "medium hair",
-        description: "orange, heavy-set, non-responsive except to neck scratches",
-        breedId: 1
-    },
-    {
-        id: 3,
-        name: "Pancake",
-        sex: "M",
-        coat: "shorthair",
-        description: "brown, heavy-set, spent life behind bars",
-        breedId: 1
-    },
-    {
-        id: 4,
-        name: "Madame Floof",
-        sex: "F",
-        coat: "long hair",
-        description: "white dark patches on legs, kerbside, fluffy dustbuster tail",
-        breedId: 2
-    },
-];
-
-
-module.exports = cats;
+// class 'CatRepository' serves as a template for creating new cat objects with props & methods
+
+class CatRepository {
+    constructor() {
+        this.cats = [ // array of objects - if object, change to >> this.cats:{ 1:{}, 2:{}, 3:{}}
+            {
+                id: 1,
+                name: "Catty",
+                sex: "F",
+                coat: "medium hair",
+                description: "buff or tan, skinny, talkative, often found in tall grasses or deep in the bush",
+                breedId: 1
+            },
+            {
+                id: 2,
+                name: "Frank",
+                sex: "M",
+                coat: "medium hair",
+                description: "orange, heavy-set, non-responsive except to neck scratches",
+                breedId: 1
+            },
+            {
+                id: 3,
+                name: "Pancake",
+                sex: "M",
+                coat: "shorthair",
+                description: "brown, heavy-set, spent life behind bars",
+                breedId: 1
+            },
+            {
+                id: 4,
+                name: "Madame Floof",
+                sex: "F",
+                coat: "long hair",
+                description: "white dark patches on legs, kerbside, fluffy dustbuster tail",
+                breedId: 2
+            },
+        ]
+    }
+    get getAllCats() {
+        return this.cats // returns [{},{},{}]
+        }
+
+    getCatByIndex (index) {
+        console.log(`retrieving cat at id: ${index}`);
+        return this.cats[index]; // returns {} (note that this.cats.index did not work)
+        }
+
+    set addCat (object) {
```

Comment:
These functions are good :) they help hide the implementation details of this class, which should make things a lot easier to change in future without `app.js` needing to alter because of those changes.

Reply:
So as I understand it,  I should move as much of the code as possible within the class, yes?

Reply:
Any code that is related to retrieving or updating cats should be in this class.